indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,309 @@
-    },
-    {
-      type: 'checkbox',
-      name: 'reservationsNeedConfirmation',
-      label: Translate.string('Require confirmation (pre-bookings)'),
-    },
-    {
-      type: 'checkbox',
-      name: 'notificationsEnabled',
-      label: Translate.string('Reminders enabled'),
-    },
-    {
-      type: 'checkbox',
-      name: 'endNotificationsEnabled',
-      label: Translate.string('Reminders of finishing bookings enabled'),
-    },
-    {
-      type: 'header',
-      label: Translate.string('Notifications'),
-    },
-    {
-      type: 'emails',
-      name: 'notificationEmails',
-      label: Translate.string('Notification emails'),
-    },
-    {
-      type: 'formgroup',
-      key: 'notifications',
-      dependsOn: 'notificationsEnabled',
-      label: Translate.string('How many days in advance booking reminders should be sent'),
-      content: [
-        {
-          type: 'input',
-          name: 'notificationBeforeDays',
-          label: Translate.string('Single/Daily'),
-          inputArgs: {
-            type: 'number',
-            min: 1,
-            max: 30,
-            fluid: true,
-          },
-          required: false,
-        },
+  const fetch = useCallback(async url => {
+    let response;
+    try {
+      response = await indicoAxios.get(url);
+    } catch (error) {
+      handleAxiosError(error);
+      return;
+    }
+    return response.data;
+  }, []);
+
+  const fetchRoomData = useCallback(async () => {
+    const resp = await Promise.all([
+      fetch(roomURL({room_id: roomId})),
+      fetch(fetchRoomAttributesURL({room_id: roomId})),
+      fetch(fetchRoomAvailabilityURL({room_id: roomId})),
+    ]);
+    [setRoomDetails, setRoomAttributes, setRoomAvailability].forEach((x, i) => x(resp[i]));
+  }, [fetch, roomId]);
+
+  const tabPanes = useMemo(
+    () =>
+      [
         {
-          type: 'input',
-          name: 'notificationBeforeDaysWeekly',
-          label: Translate.string('Weekly'),
-          inputArgs: {
-            type: 'number',
-            min: 1,
-            max: 30,
-            fluid: true,
-          },
-          required: false,
+          key: 'basic-details',
+          menuItem: <Translate>Basic Details</Translate>,
+          pane: (
+            <RoomEditDetails
+              key="basic-details"
+              favoriteUsersController={favoriteUsersController}
+            />
+          ),
+          fields: ['owner', 'key_location', 'telephone', 'capacity', 'division', 'comments'],
         },
         {
-          type: 'input',
-          name: 'notificationBeforeDaysMonthly',
-          label: Translate.string('Monthly'),
-          inputArgs: {
-            type: 'number',
-            min: 1,
-            max: 30,
-            fluid: true,
-          },
-          required: false,
+          key: 'location',
+          menuItem: <Translate>Location</Translate>,
+          pane: <RoomEditLocation key="location" />,
+          fields: [
```

Comment:
Any chance we could do some trickery with jest tests that ensures that the list here matches the final-form fields rendered by the pane?

Reply:
Yes, sure, I can do a test for that.

Reply:
After finding some problems most likely related to dependencies when testing this component, we've decided to move this to a separate task.